pkg/util/clusteragent/clusterchecks_test.go

```diff
@@ -57,6 +57,44 @@ func (suite *clusterAgentSuite) TestClusterChecksNominal() {
 	assert.Equal(suite.T(), "two", configs.Configs[1].Name)
 }
 
+// Regression test. ServiceID was not propagated to the runners and as a
+// consequence, the cluster checks rebalancing algorithm wasn't working
+// properly.
+func (suite *clusterAgentSuite) TestClusterChecksWithServiceID() {
```

Comment:
Would it make sense to add a test that setting `ServiceID` on a non-cluster agent check doesn't break things?

Reply:
non-cluster checks that run on check runners don't have this problem because they don't come from the cluster agent.

For checks in general, I just took a look and we already have several tests that set the serviceID field. For example [this one](https://github.com/DataDog/datadog-agent/blob/29fd4466c00da9f6ef1768c1337aa93cb15229cd/pkg/autodiscovery/integration/config_test.go#L225).

Reply:
That part just sets the service id explicitly. I was wondering what happens when the raw config data contains a service id where previously it couldn't, something like `pkg/autodiscovery/providers/tests/ad.yaml`.